provider/labels.go

```diff
@@ -128,3 +129,12 @@ func ensureLabelPathsExist(paths []resource.PropertyPath) tfbridge.PropertyTrans
 		return obj.ObjectValue(), nil
 	}
 }
+
+// If a resource has a `labels` field, the upstream provider will add the new aggregate label fields, `effectiveLabels`
+// and `terraformLabels`. To protect against accidental leaking of secrets via aggregation from resource-level `labels`
+// and provider-level `defaultLabels` fields, at runtime, we set these Outputs to Secret, if present.
+func setLabelsFieldsSecret(ctx context.Context, prop resource.PropertyMap) (resource.PropertyMap, error) {
+	prop["pulumiLabels"] = resource.MakeSecret(prop["pulumiLabels"])
+	prop["effectiveLabels"] = resource.MakeSecret(prop["effectiveLabels"])
```

Comment:
This doesn't apply to nested secrets. I agree with @t0yv0 that we should, do this in the schema, not at runtime.

Is there a reason why something like this doesn't work:

```patch
 provider/labels.go | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/provider/labels.go b/provider/labels.go
index 7fff7348c..12e84afa1 100644
--- a/provider/labels.go
+++ b/provider/labels.go
@@ -26,6 +26,7 @@ func fixLabelNames(prov *tfbridge.ProviderInfo) {
 	) {
 		if key == "terraform_labels" {
 			info.Name = "pulumiLabels"
+			info.Secret = tfbridge.True()
 			// Gather all paths to pulumiLabels
 			for _, v := range currentPath {
 				if v, ok := v.(string); ok && v == "*" {
@@ -36,6 +37,8 @@ func fixLabelNames(prov *tfbridge.ProviderInfo) {
 			currentPath = currentPath[:len(currentPath)-1]
 			currentPath = append(currentPath, info.Name)
 			*paths = append(*paths, currentPath)
+		} else if key == "effective_labels" {
+			info.Secret = tfbridge.True()
 		}
 
 		if m == nil {
```

Reply:
Yeah - I was thinking if we are looking to improve Secrets handling in the future, then perhaps it doesn't entirely make sense to fully schematize the Secrets here, as a fix would mean a schema change. But at that point it's six of one, half a dozen of the other.